R/route.R

```diff
@@ -410,70 +431,78 @@ route2 <- function(year, month, juveniles, inchannel_habitat, floodplain_habitat
     if (!is.null(detour)) {
       bypass <- ifelse(detour == 'sutter', "Sutter Bypass", "Yolo Bypass")
       
-      detoured_fish <- t(sapply(1:nrow(natal_watersheds$migrants), function(i) {
-        
-        rbinom(n = 4,
-               size = round(natal_watersheds$migrants[i, ]),
-               prob = proportion_flow_bypass[month, year, bypass])
-      }))
+      detoured_fish <- if (stochastic) {
+        t(sapply(1:nrow(natal_watersheds$migrants), function(i) {
+          
+          rbinom(n = 4,
+                 size = round(natal_watersheds$migrants[i, ]),
+                 prob = proportion_flow_bypass[month, year, bypass])
+        }))
+      } else {
+        round(natal_watersheds$migrants * proportion_flow_bypass[month, year, bypass])
+      }
       
       natal_watersheds$migrants <- natal_watersheds$migrants - detoured_fish
       natal_watersheds$detoured <- detoured_fish
     }
-  } else{ #if temps downstream are >18C, no fish move downstream and fish with no habitat die
+  } else { #if temps downstream are >18C, no fish move downstream and fish with no habitat die
     natal_watersheds <- fill_natal(juveniles = juveniles,
                                    inchannel_habitat = inchannel_habitat,
                                    floodplain_habitat = floodplain_habitat,
                                    territory_size,up_to_size_class = 4) #need to adjust very large fish territory size. make same as large fish?
-    number_of_regions <- max(nrow(juveniles), 1)
     
-    natal_watersheds$migrants <- natal_watersheds$detoured <- matrix(0, ncol = 4, nrow = number_of_regions)
-
-      }
-
-    return(natal_watersheds)
+    
+    natal_watersheds$migrants <- natal_watersheds$detoured <- matrix(0, ncol = 4, nrow = nrow(natal_watersheds$migrants),
```

Comment:
Instead of killing all fish above capacity we want to kill a large proportion of them and then use that proportion in the sensitivity analysis. Please change is line of code to:
  natal_watersheds$inchannel<-natal_watersheds$inchannel +  natal_watersheds$migrants*0.10 #where 0.10 is is the proportion of fish over capacity that live (what we want to vary on the logit scale in the sensitivity analysis). We add these fish to the inchannel habitat fish and then kill some again when applying inchannel survival
natal_watersheds$detoured <- matrix(0, ncol = 4, nrow = nrow(natal_watersheds$migrants),
                                                                     dimnames = dimnames(natal_watersheds$migrants))